pkg/agent/endpoints/endpoints_test.go

```diff
@@ -332,3 +330,13 @@ func logEntryWithPID(level logrus.Level, msg string, keyvalues ...interface{}) s
 	}
 	return spiretest.LogEntry{Level: level, Message: msg, Data: data}
 }
+
+func waiForListening(t *testing.T, e *Endpoints) {
+	timer := time.NewTimer(time.Second)
```

Comment:
instead of timer, may we get context and expect until it is done? (we can create a context with timeout or depends of unit test timeout) or listening is called.

Reply:
I'm taking the errCh here and getting rid of the timer.